### PR TITLE
NC-added the create a new tag functionality 

### PR DESCRIPTION
**Pull request by NathanCenter**
#### Changes Made

1. Modified file `ITagRepository.cs` to include `Tag getTagbyId(Int id), and void Add(Tag tag)` and functionality.
2. Modified file `TagRepository.cs` to include `public void add(Int id),  and public Tag getTagById(int id)` and functionality.
3. Modified file `TagController.cs` to include `IActionResult Get(int id), and public IActionResult Post(Tag tag) ` and functionality.
 1. Added file `TagCreate.js` to `Componets` directory.
 

#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout nc-addTag
    ```
4. Open a new Terminal tab (⌘T) and navigate to the server directory.
5. Test app functionality.
    When first running the code. Head to Tag management once there click on a link that says create a tag. Then you will be sent to a new page that say create a tag. Once there you type in a new tag and click the submit button. It will send you back to the Tag management page with a new tag on the list.
6. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.
Changes made after review by Fred
1.got rid of the white space
2. deleted the committed code
3. removed the unused items from the TagCreate.js
Changes made after being reviewed by Josh
1. On TagControler.cs made changes  to Tag getTagById(int id) adding PascalCasing 
2. On TagRepository.cs changed  added DbUtils to replace the cmd on the TagRepository.cs
3. On TagRepository.cs added PascalCasing to the GetTagById
4. ApplicationViews.js made changes to make it consistent with the naming conventions with teams' category route path by changing the path to a lower case tag and adding exact to the end of line 39. Also changed Line 47 to /tag/new to make everything match.
5. TagCreate.js  Added an if statement inside the handelSaveFunction that looks for an empty string and trim() to stop a user from entering in spaces.
https://github.com/NSS-Day-Cohort-53/tabloid-full-stack-always-picturesque-icebergs/projects/1#card-80770758 closed

### CODE REVIEW

File: Tabloid/client/src/components/TagCreate.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    const [tag, setTag] = useState(TagForm);
+
+    const handleInputChange = (evt) => {
+        const value = evt.target.value;
+        const key = evt.target.id;
+        const tagCopy = { ...tag };
+        tagCopy[key] = value;
+        setTag(tagCopy);
+
```
Comment (SchmidtFred) [inline Tabloid/client/src/components/TagCreate.js]: Nitpicky, but this is a lot of whitespace that could be gotten rid of to make the formatting easier for you and others to read.

File: Tabloid/client/src/modules/TagManagment.js [hunk 2]
```diff
@@ -3,15 +3,33 @@ const baseUrl = 'api/Tag';
...
+// export const addTag = (tag) => {
+//   return fetch(baseUrl, {
+//     method: "POST",
+//     headers: {
+//       "Content-Type": "application/json",
+//       Authorization: `Bearer ${token}`
+//     },
+//     body: JSON.stringify(tag),
+//   });
+// };
```
Comment (SchmidtFred) [inline Tabloid/client/src/modules/TagManagment.js]: Do we need this commented code here still?

**Review by SchmidtFred — CHANGES_REQUESTED** [review]
Good stuff Nathan! A couple of items I pointed out.

File: Tabloid/client/src/components/TagCreate.js (hunk 1 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/TagCreate.js]: @SchmidtFred  got rid of the white space

File: Tabloid/client/src/modules/TagManagment.js (hunk 2 above)
Comment (NathanCenter) [reply Tabloid/client/src/modules/TagManagment.js]: @SchmidtFred  deleted the committed code

File: Tabloid/client/src/components/TagCreate.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState  } from 'react';
+import { Form } from 'reactstrap';
+import { addTag } from '../modules/TagManagment';
+import Tags from './TagView';
+import { Link } from 'react-router-dom';
+import { useHistory } from 'react-router-dom';
```
Comment (SchmidtFred) [inline Tabloid/client/src/components/TagCreate.js]: 3 of these imported items are not used currently. Do you intend to utilize them in the future or is it okay to remove them from our import statements?

Comment (NathanCenter) [reply Tabloid/client/src/components/TagCreate.js]: @SchmidtFred removed the unused items from the TagCreate.js

**Review by SchmidtFred — APPROVED** [review]

File: Tabloid/Controllers/TagController.cs [hunk 4]
```diff
@@ -22,5 +23,24 @@ public IActionResult GetTags()
         {
             return Ok(_tagRepository.GetAllTags());
         }
+
+        [HttpPost]
+        public IActionResult Post(Tag tag)
+        {
+            _tagRepository.Add(tag);
+            return CreatedAtAction("Post", new { id = tag.Id }, tag);
```
Comment (joshdbarton) [inline Tabloid/Controllers/TagController.cs]: This should say "Get" not "Post"

File: Tabloid/Repositories/ITagRepository.cs [hunk 5]
```diff
@@ -6,5 +6,8 @@ namespace Tabloid.Repositories
     public interface  ITagRepository
     {
         List<Tag> GetAllTags();
+
+        void Add(Tag tag);
+        Tag getTagById(int id);
```
Comment (joshdbarton) [inline Tabloid/Repositories/ITagRepository.cs]: This method should be PascalCase, not camelCase named

File: Tabloid/Repositories/TagRepository.cs [hunk 6]
```diff
@@ -37,6 +38,59 @@ public List<Tag> GetAllTags()
...
+        public void Add(Tag tag)
+        {
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = @" INSERT Into Tag(Name)
+	                   OUTPUT INSERTED.ID values(@Name)";
+                    cmd.Parameters.AddWithValue("@name", tag.Name);
```
Comment (joshdbarton) [inline Tabloid/Repositories/TagRepository.cs]: Use DbUtils to do this?

File: Tabloid/Repositories/TagRepository.cs [hunk 7]
```diff
@@ -37,6 +38,59 @@ public List<Tag> GetAllTags()
...
+                    tag.Id = (int)cmd.ExecuteScalar();
+                };
+
+            }
+        }
+
+
+
+
+        public Tag getTagById(int id)
```
Comment (joshdbarton) [inline Tabloid/Repositories/TagRepository.cs]: PascalCase naming please.

File: Tabloid/client/src/components/ApplicationViews.js [hunk 8]
```diff
@@ -38,6 +39,11 @@ export default function ApplicationViews({ isLoggedIn }) {
         <Route path="/Tags">
```
Comment (joshdbarton) [inline Tabloid/client/src/components/ApplicationViews.js]: could you discuss with your team what a good naming convention strategy for your routes will be. For example, /tags/add or something like that? also make sure they are lower-cased, just to match everyone else's.

File: Tabloid/client/src/components/TagCreate.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import React, { useState  } from 'react';
+import { addTag } from '../modules/TagManagment';
+import { Link } from 'react-router-dom';
+import { useHistory } from 'react-router-dom';
+
+export const CreateTag=()=>{
+    const history = useHistory();
+    const TagForm={
```
Comment (joshdbarton) [inline Tabloid/client/src/components/TagCreate.js:9]: Is there anything in this component that prevents the user from submitting an empty tag name?

**Review by joshdbarton — CHANGES_REQUESTED** [review]

File: Tabloid/Controllers/TagController.cs (hunk 4 above)
Comment (NathanCenter) [reply Tabloid/Controllers/TagController.cs]: @joshdbarton  Changed Post to Get.

File: Tabloid/Repositories/ITagRepository.cs (hunk 5 above)
Comment (NathanCenter) [reply Tabloid/Repositories/ITagRepository.cs]: @joshdbarton changed from camelCase to PascalCase

File: Tabloid/Repositories/TagRepository.cs (hunk 6 above)
Comment (NathanCenter) [reply Tabloid/Repositories/TagRepository.cs]: @joshdbarton  added DbUtils to replace the cmd on the TagRepository.cs

File: Tabloid/Repositories/TagRepository.cs (hunk 7 above)
Comment (NathanCenter) [reply Tabloid/Repositories/TagRepository.cs]: @joshdbarton added PascalCasing to the GetTagById

File: Tabloid/client/src/components/ApplicationViews.js (hunk 8 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/ApplicationViews.js]: @joshdbarton Made it consistent with the naming conventions with teams' category route path by changing the path to a lower case tag and adding exact to the end of line 39. Also changed Line 47 to /tag/new to make everything match.

File: Tabloid/client/src/components/TagCreate.js (hunk 9 above)
Comment (NathanCenter) [reply Tabloid/client/src/components/TagCreate.js:9]: @joshdbarton  Added an if statement inside the handelSaveFunction that looks for an empty string and trim() to stop a user from entering in spaces

**Review by SchmidtFred — APPROVED** [review]
Re-approving after a misclick that removed me as reviewer.

**Review by joshdbarton — APPROVED** [review]

**Pull request merged**